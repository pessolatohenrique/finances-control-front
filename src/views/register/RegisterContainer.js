import React, { useState, useEffect } from "react";
import {
  Card,
  CardContent,
  Typography,
  Grid,
  Stepper,
  Step,
  StepLabel,
  TextField,
  Button,
  Snackbar,
  Alert,
} from "@mui/material";
import {
  THEME_COLOR,
  SNACKBAR_DIRECTION,
} from "../../constants/default_settings";
import moneyImage from "../../assets/moneyfinance-1.jpg";
import RegisterForm from "./RegisterForm";

function RegisterContainer() {
  const [step, setStep] = useState(0);
  const [formData, setFormData] = useState({});

  const steps = ["Cadastro", "Receita do Sucesso"];
  const forms = [
    <>
      <RegisterForm onColectData={colectData} />
    </>,
    <h1>componente de form 2</h1>,
  ];

  useEffect(() => {
    const formsLength = forms.length - 1;
    if (step === formsLength) {
      console.log("final form!!!");
    }
  });

  function colectData(data) {
    const updatedData = { ...formData, ...data };
    setFormData(updatedData);
    console.log("colected data!!", updatedData);
    next();
  }

  function next() {
    setStep(step + 1);
  }

  return (
    <Grid
      display="flex"
      flexDirection="column"
      flexGrow={5}
      justifyContent="center"
      alignItems="center"
      minHeight="100vh"
      sx={{
        backgroundImage: `url(${moneyImage})`,
        overflow: "hidden",
      }}
    >
      <Card sx={{ width: "50%" }}>
        <CardContent>
          <>
            <Stepper activeStep={step}>
              {[...steps].map((item, key) => (
                <Step key={key} role="custom-step">
                  <StepLabel>{item}</StepLabel>
                </Step>
              ))}
            </Stepper>
            {forms[step]}
          </>
        </CardContent>
      </Card>
    </Grid>
  );
}

export default RegisterContainer;
