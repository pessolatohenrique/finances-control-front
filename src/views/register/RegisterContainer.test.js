import {
  render,
  screen,
  fireEvent,
  waitForElement,
  waitForDomChange,
  waitFor,
} from "@testing-library/react";
import {
  REQUIRED_MESSAGE,
  PASSWORD_MISMATCH_MESSAGE,
} from "../../constants/messages";
import RegisterContainer from "./RegisterContainer";

describe("Register Container", () => {
  it("should generate snapshot", () => {
    const { container } = render(<RegisterContainer />);
    expect(container).toMatchSnapshot();
  });

  it("should validate fields when in blank", async () => {
    render(<RegisterContainer />);
    const submitBtn = screen.getByTestId("submit-button");

    fireEvent.click(submitBtn);

    await waitFor(() => expect(screen.getAllByText(REQUIRED_MESSAGE)));
  });

  it("should not register when user already exists", async () => {
    render(<RegisterContainer />);
    const username = screen.getByTestId("username");
    const email = screen.getByTestId("email");
    const password = screen.getByTestId("password");
    const confirmPassword = screen.getByTestId("confirmPassword");
    const submitBtn = screen.getByTestId("submit-button");

    fireEvent.input(username, {
      target: { value: process.env.REACT_APP_API_DEFAULT_USER },
    });

    fireEvent.input(email, {
      target: { value: process.env.REACT_APP_API_DEFAULT_EMAIL },
    });

    fireEvent.input(password, {
      target: { value: process.env.REACT_APP_API_DEFAULT_PASSWORD },
    });

    fireEvent.input(confirmPassword, {
      target: { value: process.env.REACT_APP_API_DEFAULT_PASSWORD },
    });

    fireEvent.click(submitBtn);

    await waitFor(() => expect(screen.getByText("Cadastro")));
  });

  it("should not register when password mismatch", async () => {
    render(<RegisterContainer />);
    const password = screen.getByTestId("password");
    const confirmPassword = screen.getByTestId("confirmPassword");
    const submitBtn = screen.getByTestId("submit-button");

    fireEvent.input(password, {
      target: { value: process.env.REACT_APP_API_DEFAULT_PASSWORD },
    });

    fireEvent.input(confirmPassword, {
      target: { value: "outraqualquer" },
    });

    fireEvent.click(submitBtn);

    await waitFor(() => expect(screen.getAllByText(PASSWORD_MISMATCH_MESSAGE)));
  });
});
