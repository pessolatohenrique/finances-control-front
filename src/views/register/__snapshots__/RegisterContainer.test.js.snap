// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Container should generate snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root css-136s2ky-MuiGrid-root"
  >
    &gt;
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-b2pbdi-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiStepper-root MuiStepper-horizontal css-m5vj9m-MuiStepper-root"
        >
          <div
            class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            role="custom-step"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal css-ascpo7-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    1
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-active css-qivjh0-MuiStepLabel-label"
                >
                  Cadastro
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            role="custom-step"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-disabled css-qivjh0-MuiStepLabel-label"
                >
                  Receita do Sucesso
                </span>
              </span>
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-p MuiTypography-gutterBottom css-1apbhbw-MuiTypography-root"
          >
            Preencha os campos abaixo para realizar o cadastro na plataforma
          </p>
          <form>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1rl1og4-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="username"
                    id="username-label"
                  >
                    Usuário
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      data-testid="username"
                      id="username"
                      name="username"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-dhh0px"
                      >
                        <span>
                          Usuário
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="email"
                    id="email-label"
                  >
                    E-mail
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      data-testid="email"
                      id="email"
                      name="email"
                      type="email"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-dhh0px"
                      >
                        <span>
                          E-mail
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="password"
                    id="password-label"
                  >
                    Senha
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      data-testid="password"
                      id="password"
                      name="password"
                      type="password"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-dhh0px"
                      >
                        <span>
                          Senha
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="confirmPassword"
                    id="confirmPassword-label"
                  >
                    Confirme a senha
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      data-testid="confirmPassword"
                      id="confirmPassword"
                      name="confirmPassword"
                      type="password"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-dhh0px"
                      >
                        <span>
                          Confirme a senha
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
                  data-testid="submit-button"
                  tabindex="0"
                  type="submit"
                >
                  Próximo
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
